ivy_lint/formatters/function_ordering.py

```diff
@@ -0,0 +1,134 @@
+            for node in tree.body
+        ]
+
+    def _rearrange_functions_and_classes(self, source_code: str) -> str:
+        source_code = self._remove_existing_headers(source_code)
+        
+        tree = ast.parse(source_code)
+        for node in tree.body:
+            if isinstance(node, ast.ClassDef):
+                node.body.sort(key=lambda n: n.name if isinstance(n, ast.FunctionDef) else "")
+
+        nodes_with_comments = self._extract_all_nodes_with_comments(tree, source_code)
+
+        def sort_key(item):
+            node = item[1]
+            if isinstance(node, (ast.Import, ast.ImportFrom)):
+                return (0, 0, getattr(node, "name", ""))
+            if isinstance(node, ast.Assign):
+                targets = [t.id for t in node.targets if isinstance(t, ast.Name)]
+                return (1, 0, ",".join(targets))
+            if isinstance(node, ast.ClassDef):
+                return (2, 0, node.name)
+            if isinstance(node, ast.FunctionDef):
+                if node.name.startswith("_"):
+                    return (3, 0, node.name)
+                return (4, 0, node.name)
+            return (5, 0, getattr(node, "name", ""))
+
+        nodes_sorted = sorted(nodes_with_comments, key=sort_key)
+
+        has_helper_functions = any(
+            isinstance(node, ast.FunctionDef) and node.name.startswith("_")
+            for _, node in nodes_sorted
+        )
+
+        reordered_code_list = []
+        prev_was_assignment = False
+        last_function_type = None
+
+        for code, node in nodes_sorted:
+            current_function_type = None
+            if isinstance(node, ast.FunctionDef):
+                if node.name.startswith("_"):
+                    current_function_type = "helper"
+                    if last_function_type != "helper":
+                        reordered_code_list.append("\n\n# --- Helpers --- #\n# --------------- #")
+                else:
+                    current_function_type = "api"
+                    if last_function_type != "api" and has_helper_functions:
+                        reordered_code_list.append("\n\n# --- Main --- #\n# ------------ #")
+
+
+            last_function_type = current_function_type or last_function_type
+
+            if isinstance(node, ast.Assign):
+                if prev_was_assignment:
+                    reordered_code_list.append(code.strip())
+                else:
+                    reordered_code_list.append(code)
+                prev_was_assignment = True
+            else:
+                reordered_code_list.append(code)
+                prev_was_assignment = False
+
+        reordered_code = "\n".join(reordered_code_list).strip()
+        if not reordered_code.endswith("\n"):
+            reordered_code += "\n"
+
+        return reordered_code
+
+    def _format_file(self, filename: str) -> bool:
+        if re.match(r"ivy/functional/frontends/(?!.*(?:config\.py|__init__\.py)$).*", filename) is None:
```

Comment:
I guess since we created a HEADER constant, why don't we make this as a const as well? Will improve code readability a little in the future

Reply:
Hi I have fixed this now